js/src/Axis.ts

```diff
@@ -153,16 +159,13 @@ export class Axis extends WidgetView {
                     }
                     this.axis.tickValues(useticks);
                 }
-            } else {
-                const scale = this.axis_scale.scale as (d3.ScaleLinear<number, number> |
-                                                        d3.ScaleTime<Date, number> |
-                                                        d3.ScaleLogarithmic<number, number>);
-                this.axis.tickValues(scale.ticks());
+            } else if (this.axis_scale instanceof LinearScale) {
```

Comment:
This branch doesn't include date scale or log scale?

Reply:
I added DateScale, thanks!

LogScale already has its own branch in this method.